MusinDaniilTask1/ComplexNumber.cs

```diff
@@ -0,0 +1,71 @@
+﻿
+
+
+namespace MusinDaniilTask1
+{
+    public struct Complex
+    {
+        public readonly double Re;
+        public readonly double Im;
+        public readonly double Abs;
+
+        public Complex(double re = 0, double im = 0)
+        {
+
+            Re = re;
+            Im = im;
+            Abs = Math.Sqrt((re * re) + (im * im));
+        }
+        public override int GetHashCode()
+        {
+            return ToString().GetHashCode();
+        }
+        public override bool Equals(object obj)
+        {
+            if (obj is Complex)
+            {
+                return Re == ((Complex)obj).Re & Im == ((Complex)obj).Im;
+            }
+
+            throw new ArgumentException("Объект для сравнения не является комплексным числом");
+
+        }
+        public static bool operator ==(Complex lhs, Complex rhs)
+        {
+            return lhs.Equals(rhs);
+        }
+        public static bool operator !=(Complex lhs, Complex rhs)
+        {
+            return !(lhs.Equals(rhs));
+        }
+
+        public static Complex operator +(Complex lhs, Complex rhs)
+        {
+            return new Complex(lhs.Re + rhs.Re, lhs.Im + rhs.Im);
+        }
+
+        public static Complex operator -(Complex lhs, Complex rhs)
+        {
+            return new Complex(lhs.Re - rhs.Re, lhs.Im - rhs.Im);
+        }
```

Comment:
В перегрузках операций тоже можно уменьшить инфраструктурный код
```suggestion
        public static bool operator ==(Complex lhs, Complex rhs) => lhs.Equals(rhs);

        public static bool operator !=(Complex lhs, Complex rhs) => !(lhs.Equals(rhs));

        public static Complex operator +(Complex lhs, Complex rhs) => new Complex(lhs.Re + rhs.Re, lhs.Im + rhs.Im);

        public static Complex operator -(Complex lhs, Complex rhs) => new Complex(lhs.Re - rhs.Re, lhs.Im - rhs.Im);

```